instagram-app/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import SearchBar from '../SearchBar/SearchBar';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+
+class PostsPage extends React.Component {
+    constructor(){
+        super();
+        this.state = {
+          dummyData: [], 
+          searchText: ""
+        }
+      }
+    
+      componentDidMount(){
+        this.setState({ dummyData: dummyData })
+      }
+    
+      searchHandler = (e) => {
+        e.preventDefault();
+        this.setState({ 
+          searchText: e.target.value,
+        })
+      }      
+    
+      preventRefresh = (e) => {
+        e.preventDefault();
+      }
+
+      handleLogOut = e => {
+        localStorage.removeItem('username')
+        window.location.reload();
+      }
+
+      render(){
+      return (
+        <div className="App">
+    
+          <SearchBar
+          searchHandler={this.searchHandler}
+          preventRefresh={this.preventRefresh}
+          handleLogOut={this.handleLogOut}
+          />
+    
+         <div className="card-container"> 
+          <PostContainer 
+          dummyData = {this.state.dummyData.filter(post => post.username.includes(this.state.searchText))} 
```

Comment:
I like how you accomplished the search. It's simple and no need for extra state. Keep in mind that you will be iterating over the entire dummData array every time PostContainer renders.